spec/rubocop/cop/rspec/change_by_zero_spec.rb

```diff
@@ -25,54 +25,119 @@
+        end
+      RUBY
+
+      expect_no_corrections
+    end
+
+    it 'registers an offense when the argument to `by` is zero ' \
+      'with compound expectations by `or`' do
+      expect_offense(<<-RUBY)
+        it do
+          expect { foo }
+            .to change(Foo, :bar).by(0)
+                ^^^^^^^^^^^^^^^^^^^^^^^ Prefer negated matchers with compound expectations over `change.by(0)`.
+            .or change(Foo, :baz).by(0)
+                ^^^^^^^^^^^^^^^^^^^^^^^ Prefer negated matchers with compound expectations over `change.by(0)`.
+          expect { foo }
+            .to change { Foo.bar }.by(0)
+                ^^^^^^^^^^^^^^^^^^^^^^^^ Prefer negated matchers with compound expectations over `change.by(0)`.
+            .or change { Foo.baz }.by(0)
+                ^^^^^^^^^^^^^^^^^^^^^^^^ Prefer negated matchers with compound expectations over `change.by(0)`.
+        end
+      RUBY
+
+      expect_no_corrections
+    end
+
+    it 'registers an offense when the argument to `by` is zero ' \
+      'with compound expectations by `|`' do
+      expect_offense(<<-RUBY)
+        it do
+          expect { foo }
+            .to change(Foo, :bar).by(0) |
+                ^^^^^^^^^^^^^^^^^^^^^^^ Prefer negated matchers with compound expectations over `change.by(0)`.
+                change(Foo, :baz).by(0)
+                ^^^^^^^^^^^^^^^^^^^^^^^ Prefer negated matchers with compound expectations over `change.by(0)`.
+          expect { foo }
+            .to change { Foo.bar }.by(0) |
+                ^^^^^^^^^^^^^^^^^^^^^^^^ Prefer negated matchers with compound expectations over `change.by(0)`.
+                change { Foo.baz }.by(0)
+                ^^^^^^^^^^^^^^^^^^^^^^^^ Prefer negated matchers with compound expectations over `change.by(0)`.
+        end
+      RUBY
+
+      expect_no_corrections
+    end
+  end
+
+  context "with `NegatedMatcher: 'not_change'`" do
+    let(:cop_config) { { 'NegatedMatcher' => 'not_change' } }
+
+    it 'registers an offense and autocorrect when ' \
+      'the argument to `by` is zero with compound expectations' do
+      expect_offense(<<-RUBY)
+        it do
+          expect { foo }
+            .to change(Foo, :bar).by(0)
+                ^^^^^^^^^^^^^^^^^^^^^^^ Prefer `not_change` with compound expectations over `change.by(0)`.
+            .and change(Foo, :baz).by(0)
+                 ^^^^^^^^^^^^^^^^^^^^^^^ Prefer `not_change` with compound expectations over `change.by(0)`.
+          expect { foo }
+            .to change { Foo.bar }.by(0)
+                ^^^^^^^^^^^^^^^^^^^^^^^^ Prefer `not_change` with compound expectations over `change.by(0)`.
+            .and change { Foo.baz }.by(0)
+                 ^^^^^^^^^^^^^^^^^^^^^^^^ Prefer `not_change` with compound expectations over `change.by(0)`.
+        end
+      RUBY
 
-    expect_no_corrections
+      expect_correction(<<-RUBY)
```

Comment:
Perhaps you should add examples for the single line case here, to know it would be correctly autocorrect

Reply:
Indeed it is. Added a test that is autocorrected in the case of a single line.
https://github.com/rubocop/rubocop-rspec/blob/8bddf70b7ad8ae935d8e11e339db702cddd11c79/spec/rubocop/cop/rspec/change_by_zero_spec.rb#L200-L228

I will close https://github.com/rubocop/rubocop-rspec/pull/1355.